### ISSUE TITLE
LM Studio takes up more and more memory after running for a long time

### ISSUE THREAD

**Issue by realyw** [lmstudio-ai/lmstudio-bug-tracker#172]
![clipboard_2024-10-25_15-26](https://github.com/user-attachments/assets/e2249ea3-86b8-4e67-b449-a1fbeac185fc)
I am running an embedded model, and after a long time or a large number of requests, the memory usage is very high.

**Comment by yagil** [comment]
Is this with 0.3.5?

**Comment by realyw** [comment]
> 这是 0.3.5 的吗？

Yes, it is not high at the beginning, but it gets higher as time goes by.

**Comment by realyw** [comment]
https://github.com/user-attachments/assets/b71e80f4-4668-4f8d-80b7-acea7352d0c1

**Comment by Nanyisong** [comment]
Same issue. I ran the 0.3.5 server mode (JIT loading) on a Mac Studio (M2) with the model ‘pixtral-12b’. The initial memory usage was 7.3 GB, but after two calls, the memory increased from 7.x GB to 27.49 GB.

**Comment by blast461** [comment]
I am also having this issue.   I have a 128GB of memory and running larger models in LM Studio is chewing up (leaking) memory to the point the desktop is un-usable.   I killed the LM Studio process, and it did not release the memory.   I will monitor this, since I may have been having this problem with much smaller models and the volume of RAM I have, has been hiding a bigger issue.

![Image](https://github.com/user-attachments/assets/0b042661-88be-44af-9907-38921e4d5653)
